src/main/java/gov/usds/case_issues/controllers/HitlistApiController.java

```diff
@@ -265,15 +268,29 @@ private static void assertSubparameter(String parameter, String subParameter) {
 			List<Map<String, Object>> jsonData, ZonedDateTime effectiveDate, String uploadSchema) {
 		Iterator<Map<String,Object>> valueIterator = jsonData.listIterator();
 		List<CaseRequest> newIssueCases = processCaseUploads(valueIterator, uploadSchema);
-		_uploadService.putIssueList(translated, issueTag, newIssueCases, effectiveDate);
-		return ResponseEntity.accepted().build();
+		return createUploadResponse(_uploadService.putIssueList(translated, issueTag, newIssueCases, effectiveDate));
+	}
+
+	private static ResponseEntity<?> createUploadResponse(CaseIssueUpload uploadStatus) {
+		return uploadStatus.getUploadStatus() == UploadStatus.SUCCESSFUL ?
+				ResponseEntity.accepted().build() :
+				ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
 	}
 
 	private List<CaseRequest> processCaseUploads(Iterator<Map<String, Object>> valueIterator, String schemaName) {
 		List<CaseRequest> newIssueCases = new ArrayList<>();
 		DataFormatSpec spec = _listService.getUploadFormat(schemaName);
 		try {
-			valueIterator.forEachRemaining(m -> newIssueCases.add(new MapBasedCaseRequest(spec, m)));
+			Set<String> seenReceipts = new HashSet<>();
+			while (valueIterator.hasNext()) {
+				MapBasedCaseRequest req = new MapBasedCaseRequest(spec, valueIterator.next());
+				if (!seenReceipts.add(req.getReceiptNumber())) {
+					throw new IllegalArgumentException(
+						String.format("Multiple records in input with same receipt number (%s)",
```

Comment:
Arguably better if it went through the whole list and told you how many invalid entries you had, but that seems like a you problem.